examples/cc26xx/cc26xx-ble-client-demo/client.c

```diff
@@ -0,0 +1,143 @@
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * \file
+ *    A simple IPv6-over-BLE UDP-client.
+ *
+ * \author
+ *    Michael Spoerk <michael.spoerk@tugraz.at>
+ */
+/*---------------------------------------------------------------------------*/
+#include "contiki.h"
+#include "contiki-lib.h"
+#include "contiki-net.h"
+
+#define DEBUG DEBUG_FULL
+#include "net/ip/uip-debug.h"
+#include "net/ip/uiplib.h"
+#include "net/ipv6/uip-icmp6.h"
+
+#include "cpu/cc26xx-cc13xx/ble-addr.h"
+
+#include <string.h>
+/*---------------------------------------------------------------------------*/
+#define SERVER_IP               "::"
+#define CLIENT_PORT           61617
+#define SERVER_PORT           61616
+
+#define PING_TIMEOUT              (CLOCK_SECOND / 4)
+#define CLIENT_SEND_INTERVAL      (CLOCK_SECOND * 1)
+
+#define UDP_LEN_MAX           255
+/*---------------------------------------------------------------------------*/
+static uip_ipaddr_t server_addr;
+static struct uip_icmp6_echo_reply_notification icmp_notification;
+static uint8_t echo_received;
+static struct uip_udp_conn *conn;
+
+static struct etimer timer;
+static char buf[UDP_LEN_MAX];
+static uint16_t packet_counter;
+/*---------------------------------------------------------------------------*/
+PROCESS(ipv6_ble_client_process, "IPv6 over BLE - client process");
+AUTOSTART_PROCESSES(&ipv6_ble_client_process);
+/*---------------------------------------------------------------------------*/
+void
+icmp_reply_handler(uip_ipaddr_t *source, uint8_t ttl,
+                   uint8_t *data, uint16_t datalen)
+{
+  PRINTF("echo response received\n");
+  echo_received = 1;
+}
+/*---------------------------------------------------------------------------*/
+static void
+tcpip_handler(void)
+{
+  char data[UDP_LEN_MAX];
+  if(uip_newdata()) {
+    strncpy(data, uip_appdata, uip_datalen());
+    data[uip_datalen()] = '\0';
+    PRINTF("rec. message: <%s>\n", data);
+  }
+}
+/*---------------------------------------------------------------------------*/
+static void
+timeout_handler(void)
+{
+  sprintf(buf, "Hello server %04u!", packet_counter);
+  PRINTF("send message: <%s>\n", buf);
+  uip_udp_packet_send(conn, buf, strlen(buf));
+  packet_counter++;
+}
+/*---------------------------------------------------------------------------*/
+PROCESS_THREAD(ipv6_ble_client_process, ev, data)
+{
+  uint8_t ble_addr[BLE_ADDR_SIZE];
+
+  PROCESS_BEGIN();
+  PRINTF("CC26XX-IPv6-over-BLE client started\n");
+
+  uiplib_ipaddrconv(SERVER_IP, &server_addr);
+  uip_icmp6_echo_reply_callback_add(&icmp_notification, icmp_reply_handler);
+
+  ble_addr_cpy_to((uint8_t *)&ble_addr);
```

Comment:
Why not simply `ble_addr_cpy_to(ble_addr)` - clearer and no need for the cast.

Reply:
You are right. Thanks for the hint